### PR TITLE
Add RP2350 PicoADK variant

### PR DESCRIPTION
**Pull request by DatanoiseTV**

### CODE REVIEW

File: src/boards/include/boards/datanoisetv_rp2350_dsp [hunk 1]
```diff
@@ -0,0 +1,68 @@
+ /*
```
Comment (peterharperuk) [inline src/boards/include/boards/datanoisetv_rp2350_dsp:1]: Why does this file name not end in .h?

File: src/boards/include/boards/datanoisetv_rp2350_dsp.h [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+// -----------------------------------------------------
+// NOTE: THIS HEADER IS ALSO INCLUDED BY ASSEMBLER SO
+//       SHOULD ONLY CONSIST OF PREPROCESSOR DIRECTIVES
+// -----------------------------------------------------
+//
+//------------------------------------------------------------------------------------------
+// Board definition for the DatanoiseTV RP2350 DSP Board
+//
+// This header may be included by other board headers as "boards/datanoisetv_rp2040_dsp.h"
```
Comment (lurch) [inline src/boards/include/boards/datanoisetv_rp2350_dsp.h:15]: "boards/datanoisetv_rp2350_dsp.h" ? :wink:

File: src/boards/include/boards/datanoisetv_rp2350_dsp.h [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+// -- FLASH --
+#define PICO_BOOT_STAGE2_CHOOSE_W25Q080 1
+
+#ifndef PICO_FLASH_SPI_CLKDIV
+#define PICO_FLASH_SPI_CLKDIV 2
+#endif
+
+// pico_cmake_set_default PICO_FLASH_SIZE_BYTES = (16 * 1024 * 1024)
+#ifndef PICO_FLASH_SIZE_BYTES
+#define PICO_FLASH_SIZE_BYTES (8 * 1024 * 1024)
```
Comment (lurch) [inline src/boards/include/boards/datanoisetv_rp2350_dsp.h:48]: Mismatch between 16 MB and 8 MB

File: src/boards/include/boards/datanoisetv_rp2350_dsp.h [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+#endif
+#ifndef PICO_RP2040_B0_SUPPORTED
+#define PICO_RP2040_B0_SUPPORTED 0
+#endif
```
Comment (lurch) [inline src/boards/include/boards/datanoisetv_rp2350_dsp.h:52]: Don't need this if your board uses RP2350.

File: src/boards/include/boards/datanoisetv_rp2350_dsp.h [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+// --- I2S ---
+#ifndef PICO_AUDIO_I2S_DATA_PIN
+#define PICO_AUDIO_I2S_DATA_PIN 16
+#endif
+#ifndef PICO_AUDIO_I2S_CLOCK_PIN_BASE
+#define PICO_AUDIO_I2S_CLOCK_PIN_BASE 17
+#endif
+
+#include "boards/pico.h"
```
Comment (lurch) [inline src/boards/include/boards/datanoisetv_rp2350_dsp.h:66]: a) I suspect this should be "boards/pico2.h"
b) what does this `#include` actually gain you that you haven't already defined?

**Review by lurch — DISMISSED** [review]
Marvellous, this now passes the checks I added in #1814

**Review by kilograham — APPROVED** [review]

**Review by kilograham — APPROVED** [review]

**Review by lurch — APPROVED** [review]
Only the base-branch changed, so re-approving

**Pull request merged**